增加了响度均衡的功能

423A35C7 opened the pull request:
由于每个音乐的响度不同，所以想看看有没有响度均衡的功能，看到#488 已经有人提到了，但是没有实现，于是自己实现了一下响度均衡的功能。

现在在选项中开启后，会自动读取音频中的REPLAYGAIN_TRACK_GAIN标签并调整响度，此标签是通过 [rsgain](https://github.com/complexlogic/rsgain) 预先计算后以标签的形式保存到音频文件中的，是提前用rsgain的命令行工具计算好的。以后也许可以把rsgain集成到应用中，就像修改标签、下载歌词和封面的功能那样。（或者作者也可以看看有没有别的获取REPLAYGAIN_TRACK_GAIN的方法）

### 其他
rsgain的目标标准化响度是-18 LUFS，表达的是再增加gain dB的响度就到达-18 LUFS了，但是由于平时下载的音乐的响度计算平均后为-10 LUFS（根据自己的经验得出，可能有待调整），这里要计算的是再增加多少dB的响度就到达-10 LUFS。以及BASS_FX的调整是线性音量，需要把dB转换到线性系数。

顺便夸夸作者的软件真的非常好用！希望作者能在README文档里写下贡献指南之类的，新人第一次参与开源还是有点迷茫的。

423A35C7 committed:
增加了响度均衡的功能

现在会自动读取音频中的REPLAYGAIN_TRACK_GAIN标签（使用rsgain提取）并调整响度。
rsgain的目标标准化响度是-18 LUFS，表达的是再增加gain dB的响度就到达-18 LUFS了，但是由于平时下载的音乐的响度计算平均后为-10 LUFS，这里要计算的是再增加多少dB的响度就到达-10 LUFS。

423A35C7 committed:
在选项中增加了响度均衡的复选框

423A35C7 committed:
修复了MP4文件无法读取响度均衡标签的问题

pdy265 commented:
这个功能好啊 以前千千静听上是有的，希望原作者能合并到主线里。或者我该如何下载使用你的分支呢？

zhongyang219 commented:
刚刚尝试跑了一下代码，发现CTagLibHelper::GetReplayGain函数似乎无法正确获取到REPLAYGAIN_TRACK_GAIN的值，建议使用CAudioTag::GetAudioTagPropertyMap函数获取，同时将GetReplayGain的逻辑移到CAudioTag类中。

423A35C7 committed:
响度均衡标签和其他标签一起获取，应用响度均衡的逻辑移到外层。

关于标签的逻辑：CTagLibHelper::GetReplayGain的一部分逻辑移到了CTagLibHelper::OtherPropertyToSongInfo，另一部分逻辑是只针对MP4文件的，移到了CTagLibHelper::getM4aPropertyMap。
把replay_gain属性放到了SongInfo里，同时对应修改了SongInfo::CopyAudioTag，但是没有修改CSongDataManager::SaveSongData和CSongDataManager::LoadSongData，所以现在增加新的文件时可以正常获取标签，如果已经播放过的文件需要用“重新载入播放列表”才能获取REPLAYGAIN_TRACK_GAIN标签。

应用响度均衡的逻辑：现在将ApplyReplayGain移到了CPlayer::MusicControl函数里的case Command::OPEN中，增加了IPlayerCore::ApplyReplayGain虚函数，CFfmpegCore和CMciCore里的ApplyReplayGain暂时还没有实现就先空着了。

423A35C7 committed:
修复新版应用在响度均衡上的问题

BassCore.cpp里一些多余的改动去掉；
BassFxLibrary.h里要手动添加原先bass_fx.h的部分内容；
更改选项界面，增加响度均衡的复选框。

423A35C7 committed:
修复重启应用后响度均衡会丢失的问题

在修复之前，需要手动使用“重新载入播放列表”。
这个数据版本不确定是不是应该自增一，就先这样了。
